include/zephyr/modem/pipe.h

```diff
@@ -77,7 +77,7 @@ struct modem_pipe {
  * @param data Pipe data to bind to pipe instance
  * @param api Pipe API implementation to bind to pipe instance
  */
-void modem_pipe_init(struct modem_pipe *pipe, void *data, struct modem_pipe_api *api);
+void modem_pipe_init(struct modem_pipe *pipe, void *data, const struct modem_pipe_api *api);
```

Comment:
If you want to go full circle with this, you could also add `static const` to instantiations of `struct modem_pipe_api` (in the CMUX module and the various backends).

Reply:
OK, got it.